src/app/sc2/domain/slack.py

```diff
@@ -0,0 +1,140 @@
+    }
+    info_dict = {
+        "fallback": "A match was played/closed",
+        "username": "SC2 Bot",
+    }
+    info_dict.update(get_message_data(match))
+
+    payload = urllib.urlencode(info_dict)
+
+    rpc = urlfetch.create_rpc()
+    urlfetch.make_fetch_call(
+        rpc,
+        'https://hooks.slack.com/services/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',
+        payload=payload,
+        method=urlfetch.POST,
+        headers=headers
+    )
+    try:
+        result = yield rpc
+    except Exception as e:
+        # not a huge deal if slack isn't notified, just don't bring down the rest of the request
+        logging.error(e.message)
+        result = None
+    raise ndb.Return(result)
+
+
+def get_message_data(match):
+    """
+    Returns a dict of info depending on how the match turned out
+    """
+    winner, loser = determine_match_winner_loser(match)
+    winner_names = ' & '.join([get_pretty_name(n) for n in winner])
+    loser_names = ' & '.join([get_pretty_name(n) for n in loser])
+
+    if winner and loser:
+        message = 'Congratulations on the victory {winner}!\nBetter luck next time {loser}.'.format(winner=winner_names,
+                                                                                                    loser=loser_names)
+        title = 'Match Played'
+        colour = 'good'
+        emoji = ':fallout-thumb:'
+    else:
+        match_string = get_vs_string_from_match(match)
+        message = 'Due to either inactivity or unforseen circumstances the match between {match} has been ' \
+                  'closed.'.format(match=match_string)
+        title = 'Match Closed'
+        colour = 'danger'
+        emoji = ':brucehrm:'
+
+    return {
+        'attachments': [{
+            'color': colour,
+            'fields': [{
+                'title': title,
+                'value': message,
+            }]
+        }],
+        "icon_emoji": emoji,
+    }
+
+
+def determine_match_winner_loser(match):
+    """
+    Given a match the winner and loser of the match is determined
+    :param match: The match that was played
+    :multiple return: the list of the winners of the match followed by the list of the losers of the match, will return
+    both as empty list if the match is closed without games played, will return winner as empty list and loser as list
+    of all the names if a tie
+    """
+    if match.team2_wins > match.team1_wins:
+        winner = match.team2_names
+        loser = match.team1_names
+    elif match.team1_wins > match.team2_wins:
+        winner = match.team1_names
+        loser = match.team2_names
+    else:
+        winner = []
+        loser = []
+
+    return winner, loser
+
+
+def get_vs_string_from_match(match):
+    """ Given a match and returns a formatted string that contains the players"""
+    return get_vs_string_from_names(match.team1_names, match.team2_names)
+
+
+def get_vs_string_from_names(team1, team2):
```

Comment:
Then I'd move `get_pretty_name` into the PlayerModel and then I'd have to make a property to get the actual players and put `team1_names` on the PlayerModel as well. WHERE DOES IT END!! ;p